frontend/src/pages/team/index.vue

```diff
@@ -68,21 +66,25 @@ export default {
             return this.isAdminUser || this.teamMembership?.role >= Roles.Viewer
         }
     },
+    watch: {
+        '$route.params.team_slug' (slug) {
+            this.$store.dispatch('account/setTeam', slug)
+        }
+    },
     mounted () {
         this.mounted = true
     },
     async beforeMount () {
-        await this.$store.dispatch('account/setTeam', useRoute().params.team_slug)
         this.checkRoute(this.$route)
     },
     methods: {
         checkRoute: async function (route) {
             const allowedRoutes = [
-                '/team/' + this.team.slug + '/billing',
-                '/team/' + this.team.slug + '/settings',
-                '/team/' + this.team.slug + '/settings/general',
-                '/team/' + this.team.slug + '/settings/danger',
-                '/team/' + this.team.slug + '/settings/change-type'
+                '/team/' + this.team?.slug + '/billing',
+                '/team/' + this.team?.slug + '/settings',
+                '/team/' + this.team?.slug + '/settings/general',
+                '/team/' + this.team?.slug + '/settings/danger',
+                '/team/' + this.team?.slug + '/settings/change-type'
```

Comment:
Should this not simply return `[]` if team is not set?

As it stands, you will return at array containing `'/team/undefined/billing'` etc.

Reply:
@Steve-Mcl have you tested it? I can confirm this PR fixes the bug that is live in production now that we should get fixed. I don't see any ill effects because of this.

Reply:
> have you tested it?

Yes, it works well, however I only flagged this code section because the PR (on the whole) is aboud handling against `team` being undefined. By the same virtue, if `team` IS undefined, then we will get a bunch of links to `/undefined/` 

I am not blocking it - happy for the PR to be merged as is.